kmath-jafama/src/main/kotlin/space/kscience/kmath/jafama/KMathJafama.kt

```diff
@@ -0,0 +1,102 @@
+package space.kscience.kmath.jafama
+
+import space.kscience.kmath.operations.*
+
+/**
+ * Advanced Number-like semifield that implements basic operations.
+ */
+public interface ExtendedFieldOperations<T> :
+    FieldOperations<T>,
+    TrigonometricOperations<T>,
+    PowerOperations<T>,
+    ExponentialOperations<T> {
+    public override fun tan(arg: T): T = sin(arg) / cos(arg)
+    public override fun tanh(arg: T): T = sinh(arg) / cosh(arg)
+
+    public override fun unaryOperationFunction(operation: String): (arg: T) -> T = when (operation) {
+        TrigonometricOperations.COS_OPERATION -> ::cos
+        TrigonometricOperations.SIN_OPERATION -> ::sin
+        TrigonometricOperations.TAN_OPERATION -> ::tan
+        TrigonometricOperations.ACOS_OPERATION -> ::acos
+        TrigonometricOperations.ASIN_OPERATION -> ::asin
+        TrigonometricOperations.ATAN_OPERATION -> ::atan
+        PowerOperations.SQRT_OPERATION -> ::sqrt
+        ExponentialOperations.EXP_OPERATION -> ::exp
+        ExponentialOperations.LN_OPERATION -> ::ln
+        ExponentialOperations.COSH_OPERATION -> ::cosh
+        ExponentialOperations.SINH_OPERATION -> ::sinh
+        ExponentialOperations.TANH_OPERATION -> ::tanh
+        ExponentialOperations.ACOSH_OPERATION -> ::acosh
+        ExponentialOperations.ASINH_OPERATION -> ::asinh
+        ExponentialOperations.ATANH_OPERATION -> ::atanh
+        else -> super<FieldOperations>.unaryOperationFunction(operation)
+    }
+}
+
+/**
+ * Advanced Number-like field that implements basic operations.
+ */
+public interface ExtendedField<T> : ExtendedFieldOperations<T>, Field<T>, NumericAlgebra<T>, ScaleOperations<T> {
+    public override fun sinh(arg: T): T = (exp(arg) - exp(-arg)) / 2.0
+    public override fun cosh(arg: T): T = (exp(arg) + exp(-arg)) / 2.0
+    public override fun tanh(arg: T): T = (exp(arg) - exp(-arg)) / (exp(-arg) + exp(arg))
+    public override fun asinh(arg: T): T = ln(sqrt(arg * arg + one) + arg)
+    public override fun acosh(arg: T): T = ln(arg + sqrt((arg - one) * (arg + one)))
+    public override fun atanh(arg: T): T = (ln(arg + one) - ln(one - arg)) / 2.0
+
+    public override fun rightSideNumberOperationFunction(operation: String): (left: T, right: Number) -> T =
+        when (operation) {
+            PowerOperations.POW_OPERATION -> ::power
+            else -> super<Field>.rightSideNumberOperationFunction(operation)
+        }
+}
```

Comment:
Remove it and declare kmath-core dependency (in build.gradle.kts) as follows:

```kotlin
dependencies {
  api(project(":kmath-core"))
}
```

And with Jafama it will be:

```kotlin
dependencies {
  api(project(":kmath-core"))
  api("net.jafama:jafama:2.3.2")
}
```